ankaios_sdk/_components/complete_state.py

```diff
@@ -0,0 +1,221 @@
+# Copyright (c) 2024 Elektrobit Automotive GmbH
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+This script defines the CompleteState class for managing
+the state of the system.
+
+Classes
+-------
+
+- CompleteState:
+    Represents the complete state of the system.
+
+Usage
+-----
+
+- Create a CompleteState instance:
+    .. code-block:: python
+
+        complete_state = CompleteState()
+
+- Get the API version of the complete state:
+    .. code-block:: python
+
+        api_version = complete_state.get_api_version()
+
+- Add a workload to the complete state:
+    .. code-block:: python
+
+        complete_state.set_workload(workload)
+
+- Get a workload from the complete state:
+    .. code-block:: python
+
+        workload = complete_state.get_workload("nginx")
+
+- Get a list of workloads from the complete state:
+    .. code-block:: python
+
+        workloads = complete_state.get_workloads()
+
+- Get the connected agents:
+    .. code-block:: python
+
+        agents = complete_state.get_agents()
+
+- Get the workload states:
+    .. code-block:: python
+
+        workload_states = complete_state.get_workload_states()
+"""
+
+__all__ = ["CompleteState"]
+
+from .._protos import _ank_base
+from .workload import Workload
+from .workload_state import WorkloadStateCollection
+
+
+DEFAULT_API_VERSION = "v0.1"
+
+
+class CompleteState:
+    """
+    A class to represent the complete state.
+    """
+    def __init__(self, api_version: str = DEFAULT_API_VERSION) -> None:
+        """
+        Initializes a CompleteState instance with the given API version.
+
+        Args:
+            api_version (str): The API version to set for the complete state.
+        """
+        self._complete_state = _ank_base.CompleteState()
+        self._set_api_version(api_version)
+        self._workloads: list[Workload] = []
+        self._workload_state_collection = WorkloadStateCollection()
```

Comment:
It's easier to work with a list of workloads. If the user adds one, removes one and so on. This would be the reason I kept them separate. When converting from proto, the workloads are overwritten, when converting to proto, the proto is filled with the workloads. The point would be to not change add a workload and then have to edit the proto, and when getting the workload, to convert it from the proto.

Reply:
Done. When adding configs / workloads, both proto and inner object are changed in the same place.